src/bot.js

```diff
@@ -805,5 +41,10 @@ for (const file of eventFiles) {
 	}
 }
 
-// log in to Discord with your client's token
-client.login(token);
+//  check if the Redis is ready then log in the bot to Discord
+redis.on("ready", () => {
+  console.log(`[${serverTime()}][log]: Redis is ready!`);
+
+	// log in to Discord with your client's token
+	client.login(DISCORD_BOT_TOKEN);
+});
```

Comment:
I think we can optimize this more, maybe add conditional statement from database. @Yash094

Reply:
We can ignore this for now, let's add this when we are going to optimize the code.